### PR TITLE
Scaffold functions implement scaffolding

### PR DESCRIPTION
**Pull request by fferegrino**

### CODE REVIEW

File: tests/util/test_dotted_path.py [hunk 1]
```diff
@@ -402,3 +403,24 @@ def fn(some_arg):
...
+    assert Path(tmp_directory, "sweet", "home", "__init__.py").exists()
+    assert Path(tmp_directory, "sweet", "home", "alabama.py").exists()
+
+
+def test_create_intermediate_modules_single(tmp_directory):
+    modules_and_function = ["ploomber"]
+
+    create_intermediate_modules(modules_and_function)
+
+    assert Path(tmp_directory, "ploomber.py").exists()
```
Comment (fferegrino) [inline tests/util/test_dotted_path.py:426]: We can probably combine these tests if we sacrifice some readability.

**Comment by edublancas** [comment]
Found an edge case, if a user does not pass an explicit path to a `pipeline.yaml`, scaffold calls `load_dag`:

https://github.com/ploomber/ploomber/pull/436/commits/928c5aaa1fd098eee9f8585b233559c536601bcb

Can you add a few more test cases? You can parameterize your existing tests, but do not pass the `['-e', 'pipeline.serve.yaml']` args so ploomber looks for the default `pipeline.yaml` file and it goes into this conditional:

https://github.com/ploomber/ploomber/blob/ecc5d0346db3b415c3845025aafcf2c8aff08e90/src/ploomber/cli/cli.py#L60

File: tests/cli/test_scaffold.py [hunk 2]
```diff
@@ -212,6 +213,72 @@ def test_scaffold_in_custom_entry_point(tmp_directory):
...
+    assert module_file.exists() == create_module
+    Path('pipeline.serve.yaml').write_text("""
+tasks:
+    - source: my_module.my_function
+      product: out.ipynb
+""")
+
+    runner = CliRunner()
+    result = runner.invoke(scaffold,
+                           args=['-e', 'pipeline.serve.yaml'],
```
Comment (edublancas) [inline tests/cli/test_scaffold.py:233]: please parameterize this test case, one should not pass any args so ploomber uses the default pipeline.yaml

File: tests/cli/test_scaffold.py [hunk 3]
```diff
@@ -212,6 +213,72 @@ def test_scaffold_in_custom_entry_point(tmp_directory):
...
+
+    Path('pipeline.serve.yaml').write_text(f"""
+tasks:
+    - source: {'.'.join(modules)}.{function_name}
+      product: out.ipynb
+""")
+
+    runner = CliRunner()
+    result = runner.invoke(scaffold,
+                           args=['-e', 'pipeline.serve.yaml'],
```
Comment (edublancas) [inline tests/cli/test_scaffold.py:265]: please parameterize this test case, one should not pass any args so ploomber uses the default pipeline.yaml

**Review by edublancas — CHANGES_REQUESTED** [review]
pls add test cases

File: tests/cli/test_scaffold.py (hunk 2 above)
Comment (fferegrino) [reply tests/cli/test_scaffold.py:233]: cool

File: tests/cli/test_scaffold.py (hunk 3 above)
Comment (fferegrino) [reply tests/cli/test_scaffold.py:265]: cool

File: tests/assets/spec-with-functions-no-sources/pipeline.yaml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 tasks:
-  - source: my_tasks.raw.functions.function
+  - source: my_tasks2.raw.functions.function
     product: some_file.txt
     name: raw
   
-  - source: my_tasks.clean.functions.function
+  - source: my_tasks2.clean.functions.function
     product: another.txt
```
Comment (fferegrino) [inline tests/assets/spec-with-functions-no-sources/pipeline.yaml:7]: I've changed these assets so that the tests pass, we can maybe revisit it in this ticket? https://github.com/ploomber/ploomber/issues/442

Comment (edublancas) [reply tests/assets/spec-with-functions-no-sources/pipeline.yaml:7]: I think this is a different problem. see #449

Comment (edublancas) [reply tests/assets/spec-with-functions-no-sources/pipeline.yaml:7]: but this works for now, thanks!

**Review by edublancas — APPROVED** [review]
